magicbyirineu/Modules/CardDetail/View/FavoriteButton.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  FavoriteButton.swift
+//  magicbyirineu
+//
+//  Created by andre.antonio.filho on 12/03/19.
+//  Copyright © 2019 DanielLima. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class FavoriteButton: UIButton {
+    
+    var isFavorite: Bool
```

Comment:
```suggestion
    private(set) var isFavorite: Bool
```

Se vc tem um método que faça o change pra vc, o ideal é que eu não possa alterar a variável de fora pois isso pode afetar o design da minha classe. Por exemplo, o método `setup(status: Bool)` faz alteração de appearance e título. Da forma que está eu posso alterar o `isFavorite` de fora para `true` e o estado que não é o estado verdadeiro da classe.

```suggestion
    var isFavorite: Bool {
          didSet {
               setup(status: isFavorite)
          }
    }
```

Sugiro remover o `setAsFavorite` e o `setAsNotFavorite` e criar um `setFavorite(isFavorite: Bool)`.